### PR TITLE
doc(processors): add new conditions page

### PR DESCRIPTION
**Pull request by raulb**
Final issue that fixes https://github.com/ConduitIO/conduit/issues/1422

- [x] Updates indexes on processor pages.

### CODE REVIEW

File: docs/processors/conditions.mdx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+title: 'Conditional Execution'
```
Comment (lovromazgon) [inline docs/processors/conditions.mdx:2]: We should also mention that the go template can use any function provided by [sprig](https://masterminds.github.io/sprig/).

File: docs/processors/conditions.mdx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+pipelines:
+  - id: example-pipeline
+    connectors:
+    # define source and destination connectors
+    # ...
+    processors:
+      - id: extract-name
+        plugin: json.decode
+        settings:
+          field: name
```
[the remark this answers is not in the document]
Comment (raulb) [reply docs/processors/conditions.mdx:31]: @lovromazgon oops! you're right

**Comment by raulb** [comment]
@lovromazgon thank you for your feedback. Please take another look!

File: docs/processors/conditions.mdx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+---
+title: 'Conditional Execution'
+sidebar_position: 3
+---
+
+When a [processor](/docs/connectors/getting-started) is attached to a connector or to a pipeline, we may still want to specify conditions for its execution. To do this, we simply add a `condition` key to the processor definition in our [Pipeline Configuration File](/docs/pipeline-configuration-files).
+
+The `condition` key follows the [Go templates](https://pkg.go.dev/text/template) format, allowing the use of any function provided [sprig](https://masterminds.github.io/sprig/).
```
Comment (maha-hajja) [inline docs/processors/conditions.mdx:8]:
```suggestion
The `condition` key follows the [Go templates](https://pkg.go.dev/text/template) format, allowing the use of any function provided by [sprig](https://masterminds.github.io/sprig/).
```

File: docs/processors/conditions.mdx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+title: 'Conditional Execution'
+sidebar_position: 3
+---
+
+Even when a [processor](/docs/connectors/getting-started) is attached to a connector or to a pipeline, we might want to include some conditions that need to be met for the processor to be executed at all.
+
+This can be accomplished by adding the `condition` key to the processor definition in our [Pipeline Configuration File](/docs/pipeline-configuration-files).
+
+
+:::warning
```
[the remark this answers is not in the document]
Comment (maha-hajja) [reply docs/processors/conditions.mdx:12]: I think a note bubble is good for this information, doesn't feel like a warning or something to be worried about

[the remark this answers is not in the document]
Comment (raulb) [reply docs/processors/conditions.mdx]: @maha-hajja good point. Changed to Note.

File: docs/processors/conditions.mdx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+---
+title: 'Conditional Execution'
+sidebar_position: 3
+---
+
+When a [processor](/docs/connectors/getting-started) is attached to a connector or to a pipeline, we may still want to specify conditions for its execution. To do this, we simply add a `condition` key to the processor definition in our [Pipeline Configuration File](/docs/pipeline-configuration-files).
```
Comment (maha-hajja) [inline docs/processors/conditions.mdx:6]: maybe we should mention the other option to set the `Condition` at the end of this paragraph, just a sentence saying that you can also set the condition using the HTTP API https://conditional-processors.conduit-site.pages.dev/api/#post-/v1/processors
WDYT? @lovromazgon @raulb

**Review by maha-hajja — APPROVED** [review]
nice work!

File: docs/processors/conditions.mdx (hunk 5 above)
Comment (raulb) [reply docs/processors/conditions.mdx:6]: @maha-hajja I think you're right. I should at least mention this option exists. Will do that. Thanks for the suggestion

Comment (raulb) [reply docs/processors/conditions.mdx]: @maha-hajja I included a mention to HTTP and provided better links https://github.com/ConduitIO/conduit-site/pull/65/commits/ac5aafbf05ef827c80539099d439c83794eacc31. 

Going to ship this PR, but please let me know if you think I should make a change and will address it separately. Thanks for reviewing!

Comment (maha-hajja) [reply docs/processors/conditions.mdx]: looks good!

**Pull request merged**